aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -162,16 +167,43 @@ class DispatchTable final {
     kernels_.erase(dispatch_key, operator_name_);
   }
 
+  /**
+   * Register a fallback kernel. This kernel will be returned from lookup
+   * whenever no other kernel matches the dispatch key.
+   * @param kernel Concrete kernel function implementation to register
+   */
+  void registerFallbackKernel(const DispatchTableEntry& kernel) {
+    fallback_kernel_ = kernel;
+  }
+
+  /**
+   * Deregister the fallback kernel.
+   * Without a fallback kernel, lookup of a dispatch key that doesn't match
+   * a kernel will fail again.
+   */
+  void deregisterFallbackKernel() {
+    fallback_kernel_.reset();
+  }
+
   /**
    * Perform a dynamic dispatch on this table and find the kernel to call
    * for the given arguments.
    *
    * @param args Arguments to invoke the function with
-   * @return Kernel function pointing to the right kernel for the given arguments
+   * @return Kernel function pointing to the right kernel for the given arguments.
    */
    const DispatchTableEntry& lookup(const Stack* stack) const {
      TensorTypeId dispatch_key = dispatch_strategy_.get_dispatch_key(stack);
-     return *kernels_.lookup(dispatch_key, operator_name_);
+     auto found = kernels_.lookup(dispatch_key, operator_name_);
```

Comment:
if perf matters here we also could populate all entries in the table with fallback kernel beforehand. Not sure whether it's needed or not. Benchmarks would tell (potentially in a separate PR)

Reply:
True, that's an alternative implementation. It's more complex and needs more bookkeeping since the API for registering fallback kernels could be called multiple times and needs to know which of the entries are already filled by a fallback kernel and can be replaced vs which ones have been manually registered, but we might have to go that way if it has better perf.